g2core/stepper.cpp

```diff
@@ -936,6 +936,31 @@ stat_t st_set_su(nvObj_t *nv)			// motor steps per unit (direct)
     return(STAT_OK);
 }
 
+stat_t st_set_ep(nvObj_t *nv)            // set motor enable polarity
+{
+    if (nv->value < IO_ACTIVE_LOW) { return (STAT_INPUT_LESS_THAN_MIN_VALUE); }
+    if (nv->value > IO_ACTIVE_HIGH) { return (STAT_INPUT_EXCEEDS_MAX_VALUE); }
+
+    uint8_t motor = _get_motor(nv->index);
+    if (motor > MOTORS) { return STAT_INPUT_VALUE_RANGE_ERROR; };
+
+    Motors[motor]->setMotorPolarity((ioMode)nv->value);
+    return (STAT_OK);
+}
+
+stat_t st_get_ep(nvObj_t *nv)            // get motor enable polarity
+{
+    if (nv->value < IO_ACTIVE_LOW) { return (STAT_INPUT_LESS_THAN_MIN_VALUE); }
+    if (nv->value > IO_ACTIVE_HIGH) { return (STAT_INPUT_EXCEEDS_MAX_VALUE); }
+
+    uint8_t motor = _get_motor(nv->index);
+    if (motor > MOTORS) { return STAT_INPUT_VALUE_RANGE_ERROR; };
+
+    nv->value = (float)Motors[motor]->getMotorPolarity();
+    nv->valuetype = TYPE_INT;
+    return (STAT_OK);
+}
```

Comment:
These functions do not appear to be used!

Reply:
They are used in the config_app “big table” used by the JSON and text commands.

I suspect you figured that out however due to your pull request for step polarity.

In light of that I should probably rename a few things from “motor polarity” to “enable polarity” in this pull request.

Reply:
To explain further, there are two times these get set when they’re not called explicitly via a JSON command:

In the constructor of `StepDirStepper` when the pin is setup at machine startup, then again (redundantly) when the JSON system initializes.The JSON system calls all of the table entries (that have the flags set to request it) with the default values.

Reply:
Silly me ... I didn't notice config_app diff was not expanded, so I couldn't find the calls.

Reply:
Hmm. Does this mean I should do the step polarity similarly, rather than storing it in the config as I have done?

Reply:
Well thanks for this discussion, I realized I was doing it wrong as well. I have just pushed a new change to fix where the enable pin settings are stored. Basically, it should be in the StepDirEnable (or whatever the subclasses of Stepper are that care), and be set via a `virtual` function on Stepper that can be overridden in subclasses.